### ISSUE TITLE
Declarative Hotplugging support for Volume in KubeVirt

### ISSUE THREAD

**Issue by ManigandanR2997** [kubevirt/kubevirt#13075]
****Is your feature request related to a problem? Please describe**:**

We are working on designing a GitOps Pipeline for deploying Virtual Machines in KubeVirt Clusters, for which we are using ArgoCD as a deployment tool. Initial deployment of Virtual Machine through ArgoCD works perfect. We are using HotAdd Volume feature by running virtctl commands for adding additional disks to running Virtual Machines in KubeVirt which works perfect and through --persist option we are able to make the additional disk configuration persist throughout out the Virtual Machine Lifecycle.

However, we see this as a problem as we want to manage Virtual Machine lifecycle operations for a Running VMs in a GitOps way, we would want to use ArgoCD to deploy the Virtual Machines and manage the lifecycle, but we we want to add addtional disks through Virtual Machine manifest file, the additional disk configuration do not get populated to Virtual Machine Instance.

**Describe the solution you'd like**

We would like the ability to add additional volumes to a running Kubevirt VM without requiring a restart. Specifically, we are looking for a way to dynamically attach new volumes (either through PersistentVolumeClaims or DataVolumes) while the VM is running. Ensure that these changes can be tracked and managed in manifest files through GitOps tools like ArgoCD, without needing manual intervention or the use of virtctl commands.

**Describe alternatives you've considered**:

1. Hotplugging volumes:  using virtctl commands cannot be tracked by ArgoCD, making it difficult to manage through a GitOps approach, also it complicates our deployment process and reduce automation.
2. Manifest Updates: Updating the manifest to add a new volume typically requires a restart of the VM, which we want to avoid.
 
       - dataVolume:
           hotpluggable: true
           name: datadisk-volume-hotplug-01
        name: datadisk-volume-hotplug-01
  
**Additional context**:

 This feature would make our operations much smoother. It would let us update storage without downtime and work better with our GitOps workflows.

**Comment by voatsap** [comment]
@ManigandanR2997 did you manage to fix this or if you see any workaround?

**Comment by alicefr** [comment]
There is a [PR](https://github.com/kubevirt/enhancements/pull/12) and proposal and a [PR](https://github.com/kubevirt/kubevirt/pull/13847) for adding this. In the meantime, you can always use imperative hotplig

**Comment by voatsap** [comment]
Yes we have mimic in our code virtctl logic with subresources, thank you

**Comment by fabiand** [comment]
@mhenriks IIUIC this has been addressed, not?

**Comment by mhenriks** [comment]
@ManigandanR2997 @fabiand yes, should be handled by https://github.com/kubevirt/kubevirt/pull/13847
